src/guides/relations.md

```diff
@@ -0,0 +1,562 @@
+use diesel::prelude::*;
+
+use crate::schema::books;
+use crate::schema::pages;
+
+#[derive(Queryable, Identifiable)]
+#[diesel(table_name = books)]
+pub struct Book {
+    pub id: i32,
+    pub title: String,
+}
+
+#[derive(Insertable)]
+#[diesel(table_name = books)]
+pub struct NewBook<'a> {
+    pub title: &'a str,
+}
+
+#[derive(Queryable, Identifiable, Associations)]
+#[diesel(belongs_to(Book))]
+#[diesel(table_name = pages)]
+pub struct Page {
+    pub id: i32,
+    pub content: String,
+    pub page_number: i32,
+    pub book_id: i32,
+}
+
+#[derive(Insertable)]
+#[diesel(table_name = pages)]
+pub struct NewPage<'a> {
+    pub content: &'a str,
+    pub page_number: i32,
+    pub book_id: i32,
+}
+```
+
+:::
+
+Associations in Diesel are always child-to-parent. You can declare an
+association between two records with `#[belongs_to]`. First we need to add
+`#[derive(Associations)]`, which will allow us to add
+`#[diesel(belongs_to(Book))]` to `Page` so that we can tell Diesel that pages
+belong to books and thereby reflect out 1-to-many relation.
+
+## Writing and reading data
+
+For ease of following this tutorial, we are going to put all the code into `main.rs`.
+
+::: code-block
+
+[src/main.rs]( https://github.com/fix-me/main.rs )
+
+```rust
+use diesel::pg::PgConnection;
+use diesel::prelude::*;
+use dotenvy::dotenv;
+use std::env;
+
+pub mod model;
+pub mod schema;
+
+use crate::model::*;
+use crate::schema::*;
+
+pub fn establish_connection() -> PgConnection {
+    dotenv().ok();
+
+    let database_url = env::var("DATABASE_URL").expect("DATABASE_URL must be set");
+    PgConnection::establish(&database_url)
+        .unwrap_or_else(|_| panic!("Error connecting to {}", database_url))
+}
+
+fn main() {
+    let conn = &mut establish_connection();
+
+    let new_book = NewBook {
+        title: "Momo",
+    };
+    let book: Book = diesel::insert_into(books::table)
+        .values(&new_book)
+        .get_result(conn)
+        .expect("Error saving book");
+
+    let new_page_1 = NewPage {
+        page_number: 1,
+        content: "In alten, alten Zeiten ...",
+        book_id: book.id,
+    };
+    let page_1: Page = diesel::insert_into(pages::table)
+        .values(&new_page_1)
+        .get_result(conn)
+        .expect("Error saving page 1");
+
+    let new_page_2 = NewPage {
+        page_number: 2,
+        content: "den prachtvollen Theatern...",
+        book_id: book.id,
+    };
+    let page_2: Page = diesel::insert_into(pages::table)
+        .values(&new_page_2)
+        .get_result(conn)
+        .expect("Error saving page 2");
+
+    // get pages for a book
+    let pages = Page::belonging_to(&book)
+        .inner_join(books::table)
+        .select(pages::all_columns)
+        .load::<Page>(conn)
+        .expect("Error loading pages");
+
+    // the data is the same we put in
+    assert_eq!(&page_1, pages.get(0).unwrap());
+    assert_eq!(&page_2, pages.get(1).unwrap());
+}
+```
+
+:::
+
+## many-to-many or m:n
```

Comment:
It might be useful to have a section on plain joins just between this section and the previous section?

There are a few minor things that are sometimes surprising for people new to diesel:
* Joins automatically change the default select clause
    + For inner joins: `(left::all_columns, right::all_columns)` corresponds with: `(LeftRustType, RightRustType)`
    + For left joins: `(left::all_columns, right::all_columns)` corresponds with `(LeftRustType, Option<RightRustType>)`  
   (Where the tuple `(A, B)` could also be a struct with two fields of that types, implementing `Queryable`)
* On causes are optional for joins that are based on a `joinable!` entry, but you can customize/circumvent that by just having an explicit call to `.on()` there
* For explicit select calls on queries with left joins: Any column/expression coming from the right tables needs an explicit `.nullable()` call.